### PR TITLE
Combine error and warning boxes that have the same message

### LINKED ISSUE #8151: combine duplicate error messages
It is possible for a forward model job validation to fail due to some incorrect environment setup, and forward model jobs may be called multiple times, leading to the same validation message being repeated.

To avoid cluttering up the "Some Problems Detected" panel with repeat messages, we should combine cards that have the same message but different location. The location text could show the first location and add a "and 3 more" button that can be used to view all locations where this error occurs.

### PR DESCRIPTION
**Pull request by eivindjahren**
Resolves #8151

[Screencast from 2024-06-14 12-03-22.webm](https://github.com/equinor/ert/assets/32731672/ecd54f3d-be24-4e18-8f63-df9bd900169c)

- [x] PR title captures the intent of the changes, and is fitting for release notes.
- [x] Added appropriate release note label
- [x] Commit history is consistent and clean, in line with the [contribution guidelines](https://github.com/equinor/ert/blob/main/CONTRIBUTING.md).
- [x] Make sure tests pass locally (after every commit!)

## When applicable
- [ ] **When there are user facing changes**: Updated documentation
- [x] **New behavior or changes to existing untested code**: Ensured that unit tests are added (See [Ground Rules](https://github.com/equinor/ert/blob/main/CONTRIBUTING.md#ground-rules)).
- [ ] **Large PR**: Prepare changes in small commits for more convenient review
- [ ] **Bug fix**: Add regression test for the bug
- [ ] **Bug fix**: Create Backport PR to latest release

### CODE REVIEW

File: src/ert/gui/suggestor/_suggestor_message.py [hunk 1]
```diff
@@ -58,39 +55,121 @@ def __init__(
...
-        self.setLayout(self.hbox)
+    def _toggle_expand(self, _link: Any) -> None:
+        if self._expanded:
+            self.lbl.setText(self._collapsed_text())
+            self._expand_collapse_label.setText(self._expand_link())
+            self._expanded = False
+        else:
+            self.lbl.setText(self._expanded_text())
+            self._expand_collapse_label.setText(self._hide_link())
+            self._expanded = True
```
Comment (andreas-el) [inline src/ert/gui/suggestor/_suggestor_message.py:119]: Could do even simpler, but maybe not as readable.
```self._expanded = not self._expanded```

File: src/ert/gui/suggestor/_suggestor_message.py [hunk 2]
```diff
@@ -58,39 +55,121 @@ def __init__(
...
+        location_paragraphs = ""
+        first = True
+        for loc in self._locations:
+            if first:
+                location_paragraphs += (
+                    f'<p><b style="color: {self._text_color}">location:</b>{loc}</p>'
+                )
+                first = False
+            else:
+                location_paragraphs += f"<p>{loc}</p>"
```
Comment (andreas-el) [inline src/ert/gui/suggestor/_suggestor_message.py:137]: This is also be a candidate to extract as a separate formatting function?
I am unsure if the readability is better after that though, but at least we avoid differing implementations if that would ever happen.

File: src/ert/gui/suggestor/_suggestor_message.py (hunk 1 above)
Comment (eivindjahren) [reply src/ert/gui/suggestor/_suggestor_message.py:98]: Done

File: src/ert/gui/suggestor/_suggestor_message.py [hunk 3]
```diff
@@ -58,39 +55,121 @@ def __init__(
...
+
+        return (
+            '<div style="font-size: 16px; line-height: 24px;">'
+            + f'<b style="color: {self._text_color}">'
+            + self._header
+            + "</b>"
+            + self._message
+            + location_paragraphs
+            + "</div>"
+        )
```
[the remark this answers is not in the document]
Comment (eivindjahren) [reply src/ert/gui/suggestor/_suggestor_message.py]: Extracted common text.

File: src/ert/gui/suggestor/_suggestor_message.py (hunk 2 above)
Comment (eivindjahren) [reply src/ert/gui/suggestor/_suggestor_message.py]: Not sure if I extracted the duplication that you wanted but did extract some.

**Pull request merged**